src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -153,33 +158,41 @@ export const sumTokensUiUsdBalance = (tokens: TokenUi[]): number =>
 	tokens.reduce((acc, token) => acc + (token.usdBalance ?? 0), 0);
 
 /**
- * Calculates total USD balance of mainnet tokens from the provided tokens list.
+ * Calculates total USD balance of mainnet tokens per network from the provided tokens list.
  *
  * @param $tokens - The list of tokens for filtering by network env and total USD balance calculation.
  * @param $balancesStore - The balances data for the tokens.
  * @param $exchanges - The exchange rates data for the tokens.
- * @returns The sum of mainnet tokens USD balance if balances and exchanges stores are available.
+ * @returns A NetworkId-number dictionary with total USD balance of mainnet tokens per network.
  *
  */
-export const sumMainnetTokensUsdBalance = ({
+export const calculateMainnetTokensUsdBalancesPerNetwork = ({
 	$tokens,
 	$balances,
 	$exchanges
 }: {
 	$tokens: Token[];
 	$balances: CertifiedStoreData<BalancesData>;
 	$exchanges: ExchangesData;
-}): number | undefined =>
+}): TokensTotalUsdBalancesPerNetwork =>
 	nonNullish($exchanges) && nonNullish($balances)
-		? $tokens.reduce(
-				(acc, token) =>
-					acc +
-					(token.network.env === 'mainnet'
-						? calculateTokenUsdBalance({ token, $balances, $exchanges }) ?? 0
-						: 0),
-				0
-			)
-		: undefined;
+		? $tokens.reduce<TokensTotalUsdBalancesPerNetwork>((acc, token) => {
+				if (token.network.env === 'mainnet') {
+					if (isNullish(acc[token.network.id])) {
+						acc[token.network.id] = 0;
+					}
+
+					return {
+						...acc,
+						[token.network.id]:
+							acc[token.network.id] +
```

Comment:
any loss in performance in doing here `(acc[token.network.id] ?? 0)`

Reply:
Totally! It's a one-liner now